source/IOSResolver/src/IOSResolver.cs

```diff
@@ -1808,6 +1843,7 @@ public enum CocoapodsIntegrationMethod {
                 }
                 AddPodInternal(
                     podName,
+                    "Unity-iPhone",
```

Comment:
Should this be the target name captured from line 1814?

i.e

"if (line.StartsWith("target 'Unity-iPhone' do") || line.StartsWith("target 'UnityFramework' do")) {"

I could imagine you could change this to a regex to extract the target name instead.
```
private static readonly Regex podFileTargetStartLine = new Regex("^target '([^']+)' do");
```

```
 while ((line = unityPodfile.ReadLine()) != null) {
  var currentTarget = "";
  var podFileTargetMatch = podFileTargetStartLine.Match(line);
  if (podFileTargetMatch.Success) {
    currentTarget = podFileTargetMatch.Groups[1].Value;
  }
  ...
```

You can then _potentially_ filter the target name with `XcodeTargetNames` and the target name to keep the pod in the correct target.

Reply:
This definitely makes it a lot simple to get the target. Updated to use regex.